scripts/apps/archive/controllers/file-upload-error-modal.tsx

```diff
@@ -0,0 +1,98 @@
+import React from 'react';
+
+import {gettext, gettextPlural} from 'core/utils';
+import Button from 'core/ui/components/Button';
+import {Modal} from 'core/ui/components/Modal/Modal';
+import {ModalHeader} from 'core/ui/components/Modal/ModalHeader';
+import {ModalBody} from 'core/ui/components/Modal/ModalBody';
+import {ModalFooter} from 'core/ui/components/Modal/ModalFooter';
+import {appConfig} from 'appConfig';
+
+interface IProps {
+    closeModal(): void;
+}
+
+interface IFile {
+    valid: boolean;
+    name: string;
+    width: number;
+    height: number;
+    type: string;
+}
+
+export function fileUploadErrorModal(
+    invalidFiles: Array<IFile>,
+) {
+    return class FileUploadErrorModal extends React.PureComponent<IProps> {
+        render() {
+            if (invalidFiles.length > 0) {
+                return (
+                    <Modal>
+                        <ModalHeader onClose={this.props.closeModal}>
+                            {
+                                gettextPlural(
+                                    invalidFiles.length,
+                                    'The file was not uploaded',
+                                    'Some files were not uploaded',
+                                )
+                            }
+                        </ModalHeader>
+                        <ModalBody>
+                            <h4>
+                                {
+                                    gettext(
+                                        'Minimum allowed image size is {{minWidth}}x{{minHeight}}',
+                                        {
+                                            minWidth: appConfig.pictures.minWidth,
+                                            minHeight: appConfig.pictures.minHeight,
+                                        },
+                                    )
+                                }
+                            </h4>
+                            <ol>
+                                {
+                                    invalidFiles.map((file, index) => {
+                                        if (file.type !== null) {
+                                            if (file.type.startsWith('image')) {
+                                                return (
+                                                    <li key={index}>
+                                                        {gettext('The size of {{name}} is {{width}}x{{height}}',
```

Comment:
It'd be better to call it `{{filename}}` instead of `{{name}}` so the context of the message is clear for translators.